Source/WebCore/css/ShorthandSerializer.cpp

```diff
@@ -1084,12 +1078,37 @@ String ShorthandSerializer::serializeGridTemplate() const
         return serializeLonghands(2, " / ");
     }
 
-    // FIXME: We must remove the check below and instead check that values can be represented.
-    // We only want to try serializing the interleaved areas/templates
-    // format if it was set from this shorthand, since that automatically
-    // excludes values that can't be represented in this format (subgrid,
-    // and the repeat() function).
-    if (!m_gridTemplateAreasWasSetFromShorthand)
+    // Depending on the values of grid-template-rows and grid-template-columns, we may not
+    // be able to completely represent them in this version of the grid-template shorthand.
+    // We need to make sure that those values map to a value the syntax supports
+    auto isValidTrackSize = [&] (const CSSValue& value) {
+        auto valueID = value.valueID();
+        if (valueID == CSSValueMinmax || valueID == CSSValueFitContent || valueID == CSSValueMinContent || valueID == CSSValueMaxContent || valueID == CSSValueAuto)
+            return true;
+        if (const auto* primitiveValue = dynamicDowncast<CSSPrimitiveValue>(value))
+            return primitiveValue->isLength() || primitiveValue->isPercentage() || primitiveValue->isFlex();
```

Comment:
Is `isCalculated()` representable (`calc()`)?

Reply:
Yes I think so! I made the change and added a test case that I'll upstream